test/unit/prof_small.c

```diff
@@ -0,0 +1,30 @@
+#include "test/jemalloc_test.h"
+
+TEST_BEGIN(test_profile_small_allocations) {
+	test_skip_if(!config_prof);
+	test_skip_if(!config_debug);
+
+	for (szind_t index = 0; index < SC_NBINS; index++) {
+		size_t size = sz_index2size(index);
+		assert_zu_le(size, SC_SMALL_MAXCLASS,
+		    "Unexpected large size class");
+		void *ptr = malloc(size);
+		assert_ptr_not_null(ptr, "Unexpected malloc failure");
+		edata_t *edata = emap_edata_lookup(TSDN_NULL,
+		    &arena_emap_global, ptr);
+		assert_ptr_not_null(edata,
+		    "Unable to find edata for allocation");
+		expect_false(edata_slab_get(edata),
+		    "Sampled small allocations should not be placed on slabs");
+		expect_ptr_eq(ptr, PAGE_ADDR2BASE(ptr),
+		    "Sampled allocations should be page-aligned");
```

Comment:
Let's add a few more asserts:
- edata_base_get(edata) == ptr (can probably replace the one above)
- edata_usize_get(edata) == size
- edata_size_get == PAGE_CEILING(size)
- edata_prof_tctx_get(edata)->tdata != NULL